src/Fl_BMP_Image.cxx

```diff
@@ -164,7 +166,7 @@ void Fl_BMP_Image::load_bmp_(Fl_Image_Reader &rdr)
     // If the height is negative, the row order is flipped
     temp = rdr.read_long();
     if (temp < 0) row_order = 1;
-    h(abs(temp));
+    w(ico_height > 0 ? ico_height : abs(temp));
```

Comment:
I guess this should be `h()` instead of `w()`? I think for ICO files we need to rely on the width and height data that was read from the ICO header and ignore the info from the bitmap data. So pass the width data as an argument too?
``` C++
if (ico_height > 0 && ico_width > 0) {
  rdr.read_long();
  rdr.read_long();
  w(ico_width);
  h(ico_height);
} else {
  // New BMP header...
  w(rdr.read_long());
  // If the height is negative, the row order is flipped
  temp = rdr.read_long();
  if (temp < 0) row_order = 1;
  h(abs(temp));
}